beacon-chain/core/state/transition_no_verify_sig.go

```diff
@@ -240,22 +254,43 @@ func ProcessOperationsNoVerifyAttsSigs(
 		return nil, errors.Wrap(err, "could not verify operation lengths")
 	}
 
-	state, err := b.ProcessProposerSlashings(ctx, state, signedBeaconBlock.Block().Body().ProposerSlashings(), v.SlashValidator)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not process block proposer slashings")
-	}
-	state, err = b.ProcessAttesterSlashings(ctx, state, signedBeaconBlock.Block().Body().AttesterSlashings(), v.SlashValidator)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not process block attester slashings")
-	}
-	state, err = b.ProcessAttestationsNoVerifySignature(ctx, state, signedBeaconBlock)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not process block attestations")
-	}
-	state, err = b.ProcessDeposits(ctx, state, signedBeaconBlock.Block().Body().Deposits())
-	if err != nil {
-		return nil, errors.Wrap(err, "could not process block validator deposits")
+	var err error
+	if signedBeaconBlock.Version() == version.Altair {
```

Comment:
Review this. If block is versioned altair, we call different version of slashing and attestation processing

Reply:
maybe we can have all these paths be their individual functions  and then call it here ? Rather than having a big if else statement.

Reply:
The only difference between the two code paths is `altair.SlashValidator` and `v.SlashValidator`. Therefore we can make a single function and include an additional `func` parameter.